src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -942,3 +700,222 @@ def handle_member_approval(self, slack_request: SlackActionRequest, action: str)
+                        "action_id": "static_select-action",
+                    },
+                }
+            ],
+            close={"type": "plain_text", "text": "Cancel"},
+            submit={"type": "plain_text", "text": title},
+            private_metadata=metadata,
+            callback_id=callback_id,
+        )
+
+    @abstractmethod
+    def get_modal_payload(self, callback_id: str, metadata: str) -> View:
+        raise NotImplementedError
+
+    def _update_modal(
+        self,
+        slack_client: SlackSdkClient,
+        external_id: str,
+        modal_payload: View,
+        slack_request: SlackActionRequest,
+    ) -> None:
+        try:
+            slack_client.views_update(
+                external_id=external_id,
+                view=modal_payload,
+            )
+        except SlackApiError as e:
+            # If the external_id is not found, Slack we send `not_found` error
+            # https://api.slack.com/methods/views.update
+            if unpack_slack_api_error(e) == MODAL_NOT_FOUND:
+                metrics.incr(
+                    SLACK_WEBHOOK_GROUP_ACTIONS_FAILURE_DATADOG_METRIC,
+                    sample_rate=1.0,
+                    tags={"type": "update_modal"},
+                )
+                logging_data = slack_request.get_logging_data()
+                _logger.exception(
+                    "slack.action.update-modal-not-found",
+                    extra={
+                        **logging_data,
+                        "trigger_id": slack_request.data["trigger_id"],
+                        "dialog": "resolve",
+                    },
+                )
+                # The modal was not found, so we need to open a new one
+                self._open_view(slack_client, modal_payload, slack_request)
+            else:
+                raise
+
+    def _open_view(
+        self,
+        slack_client: SlackSdkClient,
+        modal_payload: View,
+        slack_request: SlackActionRequest,
+    ) -> None:
+        # Error handling is done in the calling function
+        slack_client.views_open(
+            trigger_id=slack_request.data["trigger_id"],
+            view=modal_payload,
+        )
+
+    def open_dialog(self, slack_request: SlackActionRequest, group: Group) -> None:
+        # XXX(epurkhiser): In order to update the original message we have to
+        # keep track of the response_url in the callback_id. Definitely hacky,
+        # but seems like there's no other solutions [1]:
+        #
+        # [1]: https://stackoverflow.com/questions/46629852/update-a-bot-message-after-responding-to-a-slack-dialog#comment80795670_46629852
+        org = group.project.organization
+
+        callback_id_dict = {
+            "issue": group.id,
+            "orig_response_url": slack_request.data["response_url"],
+            "is_message": _is_message(slack_request.data),
+            "rule": slack_request.callback_data.get("rule"),
```

Comment:
This line gave me a surprising amount of trouble, trying to determine whether it was worth keeping it under the "if...channel" condition below, as is [the case](https://github.com/getsentry/sentry/pull/77454/files#diff-30aac95a95c4c9b782c0eb9a15b9d363724bc3c7a0179e1f7b65be80fbe7c7c4L438) in the original `open_archive_dialog`.

I spelunked into the Git history a bit and found that it was [originally written that way](https://github.com/getsentry/sentry/pull/63623/files#diff-30aac95a95c4c9b782c0eb9a15b9d363724bc3c7a0179e1f7b65be80fbe7c7c4R340) in #63623 by @ceorourke, and that it [was written](https://github.com/getsentry/sentry/pull/63925/files#diff-30aac95a95c4c9b782c0eb9a15b9d363724bc3c7a0179e1f7b65be80fbe7c7c4R435) to be unconditional in `open_archive_dialog` in #63925 by @isabellaenriquez. If either of you remember any significance to the distinction, let me know.

Otherwise, I'm leaning on the unconditional side, on the assumption that it's harmless for the `callback_id` to have an extra `"rule": None` entry.

Reply:
@RyanSkonnord I believe this was specific to the block kit feature flag not being GA'd at the time. The resolve dialog function was used in both our legacy alerts and the block kit alerts, but the archive dialog was only used for block kit alerts, so there wasn't a need for this conditional.

Reply:
Thanks! I'll leave it as unconditional.